client/asset/xmr/toolsdl/download.go

```diff
@@ -0,0 +1,869 @@
+// af3d98f09da94632db3e2f53c62cc612e70bf94aa5942d2a5200b4393cd9c842  monero-mac-x64-v0.18.4.4.tar.bz2
+// 7eb3b87a105b3711361dd2b3e492ad14219d21ed8fd3dd726573a6cbd96e83a6  monero-win-x64-v0.18.4.4.zip
+// a148a2bd2b14183fb36e2cf917fce6f33fb687564db2ed53193b8432097ab398  monero-win-x86-v0.18.4.4.zip
+// 84570eee26238d8f686605b5e31d59569488a3406f32e7045852de91f35508a2  monero-source-v0.18.4.4.tar.bz2
+// #
+//
+// A hashedZip struct (below) represents one line '[sha hash] [monero-<os>-<arch>-<version>.<extension>]'
+// from the above snippet.
+
+const (
+	HashesLink     = "https://www.getmonero.org/downloads/hashes.txt"
+	HashesFilename = "hashes.txt"
+	ShareDir       = "share"
+	MoneroToolsDir = "monero-tools"
+)
+
+type machine struct {
+	arch string
+	os   string
+}
+
+func getMachine() *machine {
+	return &machine{
+		os:   runtime.GOOS,
+		arch: runtime.GOARCH,
+	}
+}
+
+const (
+	CliTrigger           = "## CLI"
+	TwoSpaces            = "  "
+	Dash                 = "-"
+	Dot                  = "."
+	HashedZipsTerminator = "#"
+)
+
+const (
+	WinZipExt = ".zip"
+	TarBz2Ext = ".tar.bz2"
+)
+
+// See Also: relevant part of hashes.txt above.
+type hashedZip struct {
+	hash     string
+	zip      string
+	dir      string
+	ext      string // zip or tar.bz2
+	os       string
+	arch     string
+	valid    bool
+	selected bool
+	version  *moneroVersionV0
+}
+
+type hashedZips []hashedZip
+
+type Download struct {
+	DataDir     string
+	Log         dex.Logger
+	machine     *machine
+	tempDir     string
+	hzips       hashedZips
+	selectedZip *hashedZip
+}
+
+func (d *Download) getSelectedZip() *hashedZip {
+	return d.selectedZip
+}
+
+// getToolsBasePath gets --appdir/defaultApplicationDirectory back from DataDir
+// since we only have DataDir. Then we add the required additional  directories
+// for tools base.
+//
+// This relies on (*Core),assetDataDirectory having created the specific extra
+// three directories which we work back from so there are no validity checks.
+func (d *Download) getToolsBasePath() string {
+	dataDirPath := d.DataDir
+	assetdbDirPath := filepath.Dir(dataDirPath)
+	netDirPath := filepath.Dir(assetdbDirPath)
+	appDirPath := filepath.Dir(netDirPath)
+	return filepath.Join(appDirPath, ShareDir, MoneroToolsDir)
+}
+
+// getToolsTempPath gets base path relative to a temp dir created on Run
+func (d *Download) getToolsTempPath() string {
+	return filepath.Join(d.tempDir, ShareDir, MoneroToolsDir)
+}
+
+// ============================== local ======================================
+// GetBestCurrentLocalToolsDir retrieves latest local version directory from ToolsPath.
+// It checks any valid dir returned by getAllLocalToolsDirs to determine the latest
+// tools dir if any.
+//
+// Return: if one or more paths exist, full path, dir. Any error returned will be
+// from reading dirs and paths.
+func (d *Download) GetBestCurrentLocalToolsDir() (bool, string, string, error) {
```

Comment:
It looks like the two returned strings can just be one. The consumer can figure out the first one with the second one. They look to all be `filepath.Join(d.getToolsBasePath(), name), name`